File: mac/Tests/ClaudeToolbarCoreTests/Settings/MascotModeTests.swift
import XCTest
@testable import ClaudeToolbarCore

final class MascotModeTests: XCTestCase {
    func testNormalize() {
        XCTAssertEqual(MascotMode.normalize("full"), "full")
        XCTAssertEqual(MascotMode.normalize(" Hover "), "hover")
        XCTAssertEqual(MascotMode.normalize("OFF"), "off")
        XCTAssertEqual(MascotMode.normalize("neon"), "full")
        XCTAssertEqual(MascotMode.normalize(""), "full")
        XCTAssertEqual(MascotMode.normalize(nil), "full")
        XCTAssertEqual(MascotMode.all, ["full", "hover", "off"])
    }

    func testDefaultRoundTripAndValidation() throws {
        XCTAssertEqual(AppSettings.createDefault().behavior.mascot, "full")
        var s = AppSettings.createDefault()
        s.behavior.mascot = "off"
        let json = try SettingsJSON.encode(s)
        XCTAssertTrue(json.contains("\"mascot\""))
        XCTAssertEqual(try SettingsJSON.decode(json).behavior.mascot, "off")
        XCTAssertEqual(try SettingsJSON.decode(#"{ "behavior": {} }"#).behavior.mascot, "full")
        s.behavior.mascot = "HOVER"
        XCTAssertEqual(SettingsValidator.normalize(s).behavior.mascot, "hover")
        s.behavior.mascot = "sparkles"
        XCTAssertEqual(SettingsValidator.normalize(s).behavior.mascot, "full")
    }
}

File: tests/ClaudeToolbar.Core.Tests/Settings/MascotModeTests.cs
using ClaudeToolbar.Core.Settings;

namespace ClaudeToolbar.Core.Tests.Settings;

public class MascotModeTests
{
    [Theory]
    [InlineData("full", "full")]
    [InlineData(" Hover ", "hover")]
    [InlineData("OFF", "off")]
    [InlineData("neon", "full")]
    [InlineData("", "full")]
    [InlineData(null, "full")]
    public void NormalizeAcceptsKnownModesAndFallsBackToFull(string? input, string expected) =>
        Assert.Equal(expected, MascotMode.Normalize(input));

    [Fact]
    public void DefaultIsFull() => Assert.Equal("full", AppSettings.CreateDefault().Behavior.Mascot);

    [Fact]
    public void RoundTripsThroughJson()
    {
        var s = AppSettings.CreateDefault();
        s.Behavior.Mascot = "off";
        var json = SettingsJson.Serialize(s);
        Assert.Contains("\"mascot\": \"off\"", json);
        Assert.Equal("off", SettingsJson.Deserialize(json).Behavior.Mascot);
    }

    [Fact]
    public void MissingKeyTakesDefaultAndValidatorNormalises()
    {
        Assert.Equal("full", SettingsJson.Deserialize("{ \"behavior\": {} }").Behavior.Mascot);
        var s = AppSettings.CreateDefault();
        s.Behavior.Mascot = "HOVER";
        Assert.Equal("hover", SettingsValidator.Normalize(s).Behavior.Mascot);
        s.Behavior.Mascot = "sparkles";
        Assert.Equal("full", SettingsValidator.Normalize(s).Behavior.Mascot);
    }
}
